cmd/k8s-deployment-scaler/main.go

```diff
@@ -0,0 +1,227 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+// main initializes and runs the server, setting up TLS configuration, handling graceful shutdown, and defining HTTP routes.
+func main() {
+	tlsConfig, err := setupTLSConfig()
+	if err != nil {
+		log.Fatalf("Failed to set up TLS config: %v", err)
+	}
+
+	port := os.Getenv("SERVER_PORT")
+	if port == "" {
+		port = "8080"
```

Comment:
`8080` more commonly used for HTTP servers. 8443 would be more common for this use-case.

Reply:
Thank you.  A leftover from prior to me implementing mTLS.  I will get that updated.